generate_specs.py

```diff
@@ -187,23 +187,47 @@ def verify_one_function(func_name: str, llvm_analysis: LLVMAnalysis, out_file: P
     return success
 
 
-if __name__ == "__main__":
-    inp_file = Path(sys.argv[1])
+def _prepare_output_location(input_file_path: Path) -> Path:
+    """Return the path to the output location of the C program with generated specs.
 
-    spec_folder = Path("specs")
-    spec_folder.mkdir(exist_ok=True)
-    out_file = spec_folder / inp_file.name
+    The generated specifications sometimes make use of definitions from header files like stdlib.h
+    and limits.h. The LLM (or any other generation mechanism) likely does not include these header
+    file definitions, so we add them manually.
 
-    # Copy input file to output file initially
-    # We iteratively update the output file with specs
-    # TODO: Include <stdlib.h> and <limits.h> in out_file if not present
-    with Path(inp_file).open(encoding="utf-8") as f:
-        inp = f.read()
-    with Path(out_file).open(mode="w", encoding="utf-8") as f:
-        f.write(inp)
+    Note: The LLVM analysis should ideally expose the imports in a file, mitigating the need for the
+    brittle string matching that is currently done.
+
+    Args:
+        input_file_path (Path): The path to the initial C program for which to generate specs.
+
+    Returns:
+        Path: The path to the output location of the C program with generated specs.
+    """
+    output_folder = Path("specs")
+    output_folder.mkdir(exist_ok=True)
+    output_file_path = output_folder / input_file_path.name
+
+    with (
+        input_file_path.open(encoding="utf-8") as input_program,
+        output_file_path.open(mode="w", encoding="utf-8") as output_program,
+    ):
```

Comment:
Each file is touched only once, so you don't need a `with` block.  You can use a one-liner to read and write the file:
```python
input_program_content = inp_file.read_text()
...
output_file_path.write_text(initial_output_program)
```